### ISSUE TITLE
Add a claim that describes measurement results

### ISSUE THREAD

**Issue by laurencelundblade** [ietf-rats-wg/eat#108]
In some scenarios, it is desirable that only the positive or negative result of a measurement, possibly only a Boolean value, is reported in a token. This is in contrast to reporting the measurements (e.g., the hashes) themselves.

One scenarios is when a TEE or similar is able to conduct a full measurement and evaluation of that measurement on some other part of the system.

Another is when a Verifier checks the measurement against reference values and only reports the result to the Relying Party.

**Comment by gmandyam** [comment]
Would the following (CDDL) work?

measurement-type = {
   ? check => boolean, / Set true if measurement met criteria for pass
   ? hash => bstr(32..64) / Variable-length hash; must be minimum of 256
   }

measurement = {
     value => measurement-type
    }  / Array of measurements

This way each measurement value can provide a check, hash, both, or neither (e.g. if the measurement wasn't carried out for some reason).

**Comment by nedmsmith** [comment]
I think there should be digest claim that can be used for evidence claims (and for attestation results too). Coswid defines hash-entry which is a tuple consisting of a digest value and a hash algorithm. 

For firmware measurements (anything that doesn't have a filesystem) coswid is problematic because measurements have to be given a filesystem context. Given there is a submods context already, it doesn't make sense to require a filesystem context that requires some sort of mapping to a device context. E.g. submod-name has to match file-name or whatever.

Building hash-entry into an array of hash-entry has value for algorithm agility. A device can produce multiple digests of the same object with different algorithms. The overhead is small in CDDL. Device vendors don't have to implement algorithm agility, but it is good for those who do.

If the 'digests' type is a stand-alone claim then it can be combined with other claims (such as boolean result) or can stand-alone as a digest of attestation results. 

If there were a measurement-name claim then it would be easier to associate the name of the object with the digest of the object. submod-name could be used - it would be helpful if it was a bstr / tstr instead of just tstr so its easier to assign numeric names.

**Comment by laurencelundblade** [comment]
After a side discussion with Henk, folks from Arm and other, I took the approach of pluggable evidence formats. EAT doesn't have one format for evidence, it has a place where it can carry evidence formats defined by others.

Henk and the Arm folks are working on CoRIM and CoMID as an alternative to CoSWID. It's for the reason you mention -- CoSWID is file oriented.

I concluded that CoSWID is here to stay and must be accommodated, but that it isn't sufficient, so I took the pluggable approach. 

I don't think there's a draft for CoMID / CoRIM yet. Try here: https://github.com/ietf-rats/draft-birkholz-rats-corim/blob/master/draft-birkholz-rats-corim.md.  I don't have an opinion of whether this is good, or meets your needs or such. Maybe you want to define your own format instead. If so, it can plug into EAT. :-)

All that said, the point of this open issue is not about carrying hash values. It is about reporting to the RP whether the measurements compared correctly to the reference values. Perhaps just a simple Boolean, but probably more.
